openmc/mgxs_library.py

```diff
@@ -308,11 +319,11 @@ def name(self, name):
     @energy_groups.setter
     def energy_groups(self, energy_groups):
         # Check validity of energy_groups
-        check_type("energy_groups", energy_groups, EnergyGroups)
+        check_type("energy_groups", energy_groups, openmc.mgxs.EnergyGroups)
 
-        # Check that there is one or more groups
-        if ((energy_groups.num_groups is None) or
-            (energy_groups.num_groups < 1)):
+        # Check that there are one or more groups
+        ng = energy_groups.num_groups
```

Comment:
This error checking block is irrelevant since `EnergyGroups` can't have `None` or `< 1` for its `num_groups` attribute. Perhaps use the following replace lines 324-329:

```python
if energy_group.group_edges is None:
    msg = 'Unable to assign an EnergyGroups object '
          'with uninitialized group edges'
    raise ValueError(msg)
```